Makefile

```diff
@@ -122,6 +122,9 @@ dist/argo: vendor server/static/files.go $(CLI_PKGS)
 dist/argo-linux-amd64: vendor server/static/files.go $(CLI_PKGS)
 	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -v -i -ldflags '${LDFLAGS}' -o dist/argo-linux-amd64 ./cmd/argo
 
+dist/argo-linux-arm64: vendor server/static/files.go $(CLI_PKGS)
```

Comment:
we should use Makefile pattern here:

https://www.gnu.org/software/make/manual/html_node/Pattern-Rules.html

Reply:
Hi @alexec, thank you very much for your comments! It makes a lot of sense in simplifying the code, but still I have some tiny questions:

In current (not patched) [Makefile#L77](https://github.com/argoproj/argo/blob/d11947558bc758e5102238162036650890731ec6/Makefile#L77), we are supposed to build target `clis`, which is defined in [Makefile#L155](https://github.com/argoproj/argo/blob/d11947558bc758e5102238162036650890731ec6/Makefile#L155), and this target contains targets for every OS and architecture. I am wondering if this is what we really expect to get, as simplify those `dist/argo-OS-ARCH` targets using patterns and variables will probably change the original building behaviour, and this original behaviour is (probably) a little bit weird since it always builds targets for all platforms.

Also as it makes sense to make the code more flexible, I assume you would prefer the detection of OS and ARCH in different variables?

Thanks! I will update the PR as soon as possible once I get confirmed :)))

Reply:
Hi @alexec ! I have updated the commit based on your comments. 

From my understanding, the idea of your comments is based on simplification of the current verbosity os/arch-dependant targets. As using Makefile patterns like `dist/argo-linux-%` and `CGO_ENABLED=0 GOOS=linux GOARCH=% go build -v -i -ldflags '${LDFLAGS}' -o dist/argo-linux-% ./cmd/argo` will lead to compiler error: `cmd/go: unsupported GOOS/GOARCH pair linux/%`, I choosed using `GOOS` and `GOARCH` variables. Please take a look and give comments if you like :)))

Reply:
Found a way to use Makefile pattern, updated the code and resolving this conversation.